llpc/tool/llpcPipelineBuilder.cpp

```diff
@@ -74,171 +73,28 @@ namespace Llpc {
 namespace StandaloneCompiler {
 
 // =====================================================================================================================
-// Builds pipeline using the provided build info and performs linking.
+// Factory function that returns a `PipelineBuilder` appropriate for the given pipeline type (e.g., graphics, compute).
+// To support new pipeline types, create a new class dering from `PipelineBuilder` and register its constructor here.
 //
-// @returns : Result::Success on success, other status on failure.
-Result PipelineBuilder::build() {
-  // Even though `build*Pipeline` produces a single `BinaryData` today, future pipeline types () will
-  // result multiple outputs. We use a vector instead of a single variable to prepare for that.
-  SmallVector<BinaryData, 1> pipelines;
-  pipelines.push_back({});
-  const bool isGraphics = isGraphicsPipeline(m_compileInfo.stageMask);
-  const bool isCompute = isComputePipeline(m_compileInfo.stageMask);
-  assert(!(isGraphics && isCompute) && "Bad stage mask");
-
-  if (isGraphics) {
-    Result result = buildGraphicsPipeline(pipelines[0]);
-    if (result != Result::Success)
-      return result;
-  } else if (isCompute) {
-    Result result = buildComputePipeline(pipelines[0]);
-    if (result != Result::Success)
```

Comment:
Using `llvm::Error`/`llvm::Expected<T>` makes it easier to propagate errors without having to use output parameters and require callers to properly initialize them.

In LLPC, we have a custom dynamic error type `llpc::ResultError` that bridges `llvm::Error` and `Vkgc::Result`: https://github.com/GPUOpen-Drivers/llpc/blob/dev/docs/CodingStandards.md#internal-apis.
The mapping is:
*  `Result::Success` --> `Error::Success`
*  Other `Result::Xyz` -> `createResultError(Result::Xyz)`

Since this PR changes the API without changing the logic, I thought that this transition should be fine. If you think that this is a bit too much at once and we should do it in two steps (i.e., first preserve `Result`-based error handling, and then have a follow-up clean up PR), I can do that.